### ISSUE TITLE
Duplicate log entries when running a ioloop inside cement

### ISSUE THREAD

**Issue by macnibblet** [datafolklabs/cement#310]
Heya

I have a slight misc issue with duplicated log entries after initiating a tornado ioloop inside of cement. I have no idea whats causing this tho, just wondering if anyone has any suggestions ?

**Comment by derks** [comment]
Sounds odd...  any chance you can provide an example Cement app that demonstrates the issues (as little code as possible to reproduce)?

**Comment by derks** [comment]
@macnibblet sorry but I can't really proceed without some sample code to reproduce the issue.

**Comment by macnibblet** [comment]
Sorry, forgot about this issue, ill try and produce a test case on monday.

**Comment by macnibblet** [comment]
Actually it has nothing to do with the ioloop, but just importing logging and using it from their with code that was not originally designed to run in cement causes the double logging.

**Comment by derks** [comment]
Thanks for the followup.  If you can provide a [MCVE](http://stackoverflow.com/help/mcve) it would be helpful, otherwise I have little to go on in order to verify what you are seeing.

**Comment by derks** [comment]
@macnibblet any word on this?

**Comment by derks** [comment]
Closing as there has been no further feedback provided.

**Comment by coccoinomane** [comment]
Hell,
I understand the issue of duplicate log entries will be fixed in 3.0.10.  In the meanwhile, I am employing this quick fix..

1. Define a function that kills the output of the root logger:
   ```
   def silence_root_logger() -> None:
       import logging
   
       logging.getLogger().handlers = []
   ```
2. Attach this function to the `pre_setup` hook.  This goes in the `Meta` class inside your app class:
   ```
   # extend the app with cement hook system
   hooks = [
        ("pre_setup", lambda app: silence_root_logger()),
   ]
   ```

Cheers,
Cocco

**Comment by derks** [comment]
@coccoinomane thank you for that!  I just found something else that works builtin:

- https://stackoverflow.com/questions/19561058/duplicate-output-in-simple-python-logging-configuration/19561320#19561320
- https://docs.python.org/3/library/logging.html#logging.Logger.propagate

Merged to `main`, and will be released in `stable/3.0.10`.

**Comment by coccoinomane** [comment]
Thank you @derks for fixing the issue!
I am also grateful for the improvements to cement you made during the last few weeks 🙏
Cheers,
Cocco
